system/Router/Router.php

```diff
@@ -614,29 +632,53 @@ protected function validateRequest(array $segments): array
 	/**
 	 * Sets the sub-directory that the controller is in.
 	 *
-	 * @param string|null   $dir
-	 * @param boolean|false $append
+	 * @param string|null $dir
+	 * @param boolean     $append
+	 * @param boolean     $validate if true, checks to make sure $dir consists of only PSR4 compliant segments
 	 */
-	public function setDirectory(string $dir = null, bool $append = false)
+	public function setDirectory(string $dir = null, bool $append = false, bool $validate = true)
 	{
 		if (empty($dir))
 		{
 			$this->directory = null;
-			return;
+			return true;
```

Comment:
Why returns `true`?

Reply:
Thank you so much for reviewing the code closely!

> Why returns `true`?

Oh no that's a mistake. I had considered having setDirectory return a boolean value if the directory was valid. I changed my mind but accidentally left this in there. I will remove the return value of true.